docs/ProfilingDevelopment.md

```diff
@@ -0,0 +1,91 @@
+# Profiling Development
+
+## What?
+
+This file contains development notes specific to the profiling feature.
+
+For a more practical view of getting started with development of `ddtrace`, see <DevelopmentGuide.md>.
+
+## Profiling components high-level view
+
+Components below live inside <../lib/ddtrace/profiling>:
```

Comment:
I like the list and description of the main components.

If we don't already, we should ensure such descriptions are present in each of the relevant definitions within the code, and that these descriptions are consistent with an existing ones.

Side note; downside to this is if we do any refactoring, this is the mostly likely thing to fall out of date and require updating.

Reply:
I've made sure the docs are in sync with the components in 8b33e60282c8a5013e5b362140a58a20cb4f630a .

> Side note; downside to this is if we do any refactoring, this is the mostly likely thing to fall out of date and require updating.

Indeed. I still think these docs are a valuable entry-point to someone exploring the codebase, as they try to give a glimpse of the forest, rather than just the trees :)